utils/gen_xpub.go

```diff
@@ -0,0 +1,83 @@
+// Copyright ©, 2024-present, Lightspark Group, Inc. - All Rights Reserved
+package utils
+
+import (
+	"encoding/hex"
+	"log"
+
+	// "flag"
+
+	"github.com/btcsuite/btcd/btcutil/hdkeychain"
+	"github.com/btcsuite/btcd/chaincfg"
+)
+
+func getBitcoinNetwork(network string) (uint32, *chaincfg.Params) {
+	switch network {
+	case "mainnet":
+		return 0, &chaincfg.MainNetParams
+	case "testnet":
+		return 1, &chaincfg.TestNet3Params
+	case "regtest":
+		return 1, &chaincfg.RegressionNetParams
+	default:
+		log.Fatalf("Network not supported: %s", network)
+		return 0, nil
+	}
+}
+
+func GenHardenedXPub(masterSeedHex string, bitcoinNetwork string) (string, error) {
+	networkID, params := getBitcoinNetwork(bitcoinNetwork)
+	masterSeed, err := hex.DecodeString(masterSeedHex)
+	if err != nil {
+		log.Fatalf("Failed to decode master seed from hex: %v", err)
+	}
+
+	masterKey, err := hdkeychain.NewMaster(masterSeed, params)
+	if err != nil {
+		log.Fatalf("Failed to create master key: %v", err)
+	}
+
+	accountKey, err := masterKey.Derive(hdkeychain.HardenedKeyStart + 84)
+	if err != nil {
+		log.Fatalf("Failed to derive key: %v", err)
+	}
+
+	purposeKey, err := accountKey.Derive(hdkeychain.HardenedKeyStart + networkID)
+	if err != nil {
+		log.Fatalf("Failed to derive key: %v", err)
+	}
+
+	changeKey, err := purposeKey.Derive(hdkeychain.HardenedKeyStart + 0)
+	if err != nil {
+		log.Fatalf("Failed to derive key: %v", err)
+	}
+
+	xpub, err := changeKey.Neuter()
+	if err != nil {
+		log.Fatalf("Failed to neuter key: %v", err)
+	}
+	return xpub.String(), nil
+}
+
+// Uncomment to run as an executable script for customers
```

Comment:
Hmm, maybe move this function to the example folder or create a script folder for this. I don't think we want commented code in the SDK itself.

Reply:
Makes sense, will add that now

Reply:
![Screenshot 2024-10-29 at 5.01.53 PM.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/EJphitdgYFvuACNazm1q/5c881d2b-8b14-42c3-ac85-f6647d2c5951.png)